### PR TITLE
planner: fix group_concat function couldn't resolve the index of its order-by item

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/41986

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
planner: fix group_concat function couldn't resolve the index of its order-by item
```

### CODE REVIEW

File: planner/core/rule_aggregation_push_down.go [hunk 1]
```diff
@@ -571,6 +574,26 @@ func (a *aggregationPushDownSolver) aggPushDown(p LogicalPlan, opt *logicalOptim
 							break
 						}
 						newAggFuncsArgs = append(newAggFuncsArgs, newArgs)
+						if len(aggFunc.OrderByItems) != 0 {
```
Comment (qw4990) [inline planner/core/rule_aggregation_push_down.go:577]: Please add some comments for this part.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**